### PR TITLE
Make basic test framework for examples

### LINKED ISSUE #42: Create test framework for examples

### PR DESCRIPTION
**Pull request by mhorod**
Resolves #42

### CODE REVIEW

**Comment by mhorod** [comment]
There are few depenedncies on other here like FileInput and Lexer itself, so we should imo we can merge this one at the end when we have the whole process configured

File: src/test/kotlin/cacophony/examples/ExampleRunner.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
+package cacophony.examples
+
+import cacophony.utils.Input
+import cacophony.utils.Location
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Path
+import kotlin.io.path.isRegularFile
+
+interface Runner {
```
Comment (Giselus) [inline src/test/kotlin/cacophony/examples/ExampleRunner.kt:10]: Maybe still ExampleRunner?

File: src/test/kotlin/cacophony/examples/ExampleRunner.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        input: Input,
+        diagnostics: TestDiagnostics,
+    )
+}
+
+interface DiagnosticsAssertion {
+    fun check(diagnostics: TestDiagnostics)
+}
+
+class FileInput(path: Path) : Input {
```
Comment (Giselus) [inline src/test/kotlin/cacophony/examples/ExampleRunner.kt:21]: FileInput is already merged, please change.

File: examples/incorrect/assignment/descriptions.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{
+  "examples": [
+    {
+      "path": "assignment-incorrect-types.cac",
+      "errors": {
+        "lexerErrors": false
+      }
+    },
+    {
+      "path": "descriptions.json",
```
Comment (Giselus) [inline examples/incorrect/assignment/descriptions.json:10]: I guess this file shouldn't contain it's own description?

**Review by Giselus — CHANGES_REQUESTED** [review]
I'm not a fan of current solution. We should have more precise error, not only information if it occurs or not.

**Comment by Giselus** [comment]
I guess there will be few more things to fix, but we'll get back to it after other things are merged.

File: examples/incorrect/assignment/descriptions.json (hunk 3 above)
Comment (mhorod) [reply examples/incorrect/assignment/descriptions.json:10]: Nice catch, I didn't even notice, will fix

**Comment by mhorod** [comment]
> I'm not a fan of current solution. We should have more precise error, not only information if it occurs or not.

Yeah, I agree, although we didn't really discuss error handling. We should  get back to this when lexer gets implemented

**Review by Giselus — APPROVED** [review]

**Pull request merged**